lib/ansible/modules/packaging/os/yum.py

```diff
@@ -1439,7 +1439,12 @@ def ensure(self, repoq):
                 if self.enablerepo:
                     try:
                         for rid in self.enablerepo:
-                            my.repos.enableRepo(rid)
+                            try:
+                                my.repos.enableRepo(rid)
+                            except yum.Errors.YumBaseError as e:
+                                if to_text('repository not found') in to_text(e):
```

Comment:
I *think* avoiding `to_text` call by doing `u'repository not found' in to_text(e)` instead might be better?